opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -567,10 +569,32 @@ private static void createObservation(Event e, JSONObject jsonObject, String val
             e.addObs(new Obs(CONCEPT, dataType, entityIdVal, entityParentVal, vall, humanReadableValues, null,
                     formSubmissionField));
         } else if (StringUtils.isBlank(entityVal)) {
+
             vall.add(obsValue);
 
-            e.addObs(new Obs("formsubmissionField", dataType, formSubmissionField, "", vall, new ArrayList<>(), null,
-                    formSubmissionField));
+            if ((AllConstants.NATIVE_RADIO.equals(widgetType) || AllConstants.EXTENDED_RADIO_BUTTON.equals(widgetType)) &&
+                    jsonObject.has(AllConstants.OPTIONS)) {
+                Map<String, Object> keyValPairs = new HashMap<>();
+                try {
+                    JSONArray options = getJSONArray(jsonObject, AllConstants.OPTIONS);
+                    for (int i = 0; i < options.length(); i++) {
+                        JSONObject option = options.getJSONObject(i);
+                        if (obsValue.equals(option.getString(KEY))) {
+                            keyValPairs.put(obsValue, option.optString(TEXT));
+                            break;
+                        }
```

Comment:
The logic here seems similar to https://github.com/OpenSRP/opensrp-client-core/blob/2bb560c829b13394ab3389bd9f98649f90f5d0c6/opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java#L419. Is there a way to merge the two?

Reply:
This will not be convenient since Check Box loops checking on options `VALUE` while Native Radio checks on options `KEY` , native radio does not have `VALUE` attribute in its option object and it does not need a full traversal, it stops looping when it finds a match.

Reply:
cool